tests/core/featurizers/test_single_state_featurizers.py

```diff
@@ -183,143 +141,145 @@ def test_single_state_featurizer_creates_encoded_all_actions():
-                },
-            ],
-        },
-        interpreter=interpreter,
-    )
-    assert sorted(list(encoded.keys())) == sorted([ENTITY_TAGS])
-    assert np.all(
-        encoded[ENTITY_TAGS][0].features == [[0], [0], [0], [0], [1], [0], [2]]
-    )
 
+def sparse_equals_dense(
+    sparse_matrix: scipy.sparse.spmatrix, dense_matrix: np.ndarray
+) -> bool:
+    return (sparse_matrix != scipy.sparse.coo_matrix(dense_matrix)).nnz == 0
 
-@pytest.mark.timeout(
-    300, func_only=True
-)  # these can take a longer time than the default timeout
-def test_single_state_featurizer_with_bilou_entity_roles_and_groups(
-    unpacked_trained_spacybot_path: Text,
-):
-    from rasa.core.agent import Agent
 
-    interpreter = Agent.load(unpacked_trained_spacybot_path).interpreter
-    # TODO roles and groups are not supported in e2e yet
-    domain = Domain(
-        intents=[],
-        entities=["city", f"city{ENTITY_LABEL_SEPARATOR}to"],
-        slots=[],
-        responses={},
-        forms={},
-        action_names=[],
-    )
+@pytest.mark.parametrize("action_name", ["None", "not_action_listen", "action_listen"])
+def test_encode_state__without_lookup(action_name: bool):
+    """Tests that `encode_state` creates features for every attribute.
+    In particular, that this is done even when there is no lookup table.
+    If there is no action_listen in the  state, then no features should be created for
+    the user sub-state.
+    """
     f = SingleStateFeaturizer()
-    f.prepare_for_training(domain, RegexInterpreter(), bilou_tagging=True)
+    f._default_feature_states[INTENT] = {"a": 0, "b": 1}
+    f._default_feature_states[ACTION_NAME] = {"c": 0, "d": 1, "action_listen": 2}
+    f._default_feature_states[SLOTS] = {"e_0": 0, "f_0": 1, "g_0": 2}
+    f._default_feature_states[ACTIVE_LOOP] = {"h": 0, "i": 1, "j": 2, "k": 3}
 
-    encoded = f.encode_entities(
-        {
-            TEXT: "I am flying from London to Paris",
-            ENTITIES: [
-                {
-                    ENTITY_ATTRIBUTE_TYPE: "city",
-                    ENTITY_ATTRIBUTE_VALUE: "London",
-                    ENTITY_ATTRIBUTE_START: 17,
-                    ENTITY_ATTRIBUTE_END: 23,
-                },
-                {
-                    ENTITY_ATTRIBUTE_TYPE: f"city{ENTITY_LABEL_SEPARATOR}to",
-                    ENTITY_ATTRIBUTE_VALUE: "Paris",
-                    ENTITY_ATTRIBUTE_START: 27,
-                    ENTITY_ATTRIBUTE_END: 32,
-                },
-            ],
-        },
-        interpreter=interpreter,
-        bilou_tagging=True,
-    )
-    assert sorted(list(encoded.keys())) == sorted([ENTITY_TAGS])
-    assert np.all(
-        encoded[ENTITY_TAGS][0].features == [[0], [0], [0], [0], [4], [0], [8]]
+    # Set the action_name properly...
+    if action_name == "not_action_listen":
+        action_name = "d"
+    elif action_name == "None":
+        action_name = None
+    else:
+        assert action_name == "action_listen"
```

Comment:
thought it might be easier to understand the then, just didn't think of just using the passed on "not_action_listen" in the example :D - will rework this, thanks :)

Reply:
done